core/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -184,4 +190,27 @@ private void failIfOverShardCountLimit(ClusterService clusterService, int shardC
         }
     }
 
+    private void resolveIndexBoost(SearchRequest searchRequest, ClusterState clusterState) {
+        if (searchRequest.source() == null) {
+            return;
+        }
+
+        List<Tuple<String, Float>> indexBoost = searchRequest.source().indexBoost();
+        if (indexBoost == null) {
+            return;
+        }
+
+        List<Tuple<String, Float>> concreteIndexBoost = new ArrayList<>();
+        for (Tuple<String, Float> ib : indexBoost) {
```

Comment:
Right.
I didn't like tuple since it's not very readable as you pointed out. So, I decided to create a new class.